### PR TITLE
[GLUTEN-587] Refactor Gluten ExpressionConverter - Part One - Simplify ExpressionConverter, remove repetitive code

### PR DESCRIPTION
**Pull request by zzcclp**
## What changes were proposed in this pull request?
Simplify ExpressionConverter, remove repetitive code.

After this pr, when add a new expression, only needs to add a mapping into `ExpressionMappings.SCALAR_SIGS`, for example :
`Sig[Asin](ASIN)  // **Asin** is a Spark expression, **ASIN** is a substrait expression name `.

But if it needs to specially implement a expression, must add a new `ExpressionTransformer` implemented class, and then add in `ExpressionConverter.replaceWithExpressionTransformer`. Please refer to `CaseWhenTransformer`.

Now it also can support to add a black list of the scalar function and aggregation function for each backend, please refer to `CHExpressionUtil.CH_EXPR_BLACKLIST` and `CHExpressionUtil.CH_AGGREGATE_FUNC_BLACKLIST`.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by lviiii** [comment]
Velox backend built successfully on local env and passed TPCH.

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionMappings.scala [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    Sig[Min](MIN),
+    Sig[Max](MAX),
+    Sig[StddevSamp](STDDEV_SAMP)
+  )
+
+  lazy val scalar_functions_map: Map[Class[_], String] =
+    SCALAR_SIGS.map(s => (s.expClass, s.name)).toMap
+  lazy val aggregate_functions_map: Map[Class[_], String] =
+    AGGREGATE_SIGS.map(s => (s.expClass, s.name)).toMap
+}
```
Comment (lviiii) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionMappings.scala:300]: Velox backend built successfully on local env and passed TPCH.

File: backends-clickhouse/src/main/scala/io/glutenproject/utils/CHExpressionUtil.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    LPAD -> Set(EMPTY_TYPE),
+    RPAD -> Set(EMPTY_TYPE),
+    REVERSE -> Set(EMPTY_TYPE),
+    SPLIT -> Set(EMPTY_TYPE),
+    PMOD -> Set(EMPTY_TYPE),
+    GET_JSON_OBJECT -> Set(EMPTY_TYPE),
+    JSON_ARRAY_LENGTH -> Set(EMPTY_TYPE),
+    MURMUR3HASH -> Set(EMPTY_TYPE),
+    MD5 -> Set(EMPTY_TYPE),
+    SPLIT_PART -> Set(EMPTY_TYPE)
```
Comment (lviiii) [inline backends-clickhouse/src/main/scala/io/glutenproject/utils/CHExpressionUtil.scala:46]: @taiyang-li Could you add CH expressions which caused failed Spark UT into this blacklist? Failed UT refer to https://github.com/oap-project/gluten/pull/643.

**Review by FelixYBW — APPROVED** [review]
Thank you for your work. Code is much cleaner now.

**Review by zhztheplayer — APPROVED** [review]
👍

**Pull request merged**